erpnext/stock/stock_ledger.py

```diff
@@ -616,24 +619,25 @@ def process_sle_against_current_timestamp(self):
 			self.process_sle(sle)
 
 	def get_sle_against_current_voucher(self):
-		self.args["time_format"] = "%H:%i:%s"
+		self.args["posting_datetime"] = get_combine_datetime(
+			self.args.posting_date, self.args.posting_time
+		)
 
 		return frappe.db.sql(
 			"""
 			select
-				*, timestamp(posting_date, posting_time) as "timestamp"
+				*, posting_datetime as "timestamp"
 			from
 				`tabStock Ledger Entry`
 			where
 				item_code = %(item_code)s
 				and warehouse = %(warehouse)s
 				and is_cancelled = 0
 				and (
-					posting_date = %(posting_date)s and
-					time_format(posting_time, %(time_format)s) = time_format(%(posting_time)s, %(time_format)s)
+					posting_datetime = %(posting_datetime)s
```

Comment:
This isn't same as before. The previous code was stripping milliseconds. We can convert this into two different conditions

```
posting_datetime >= time.replace(millis=0) 
AND 
posting_date < time.replace(mills=0) + timedelta(seconds=0)
```

Reply:
I have already tried this but the test case test_timestamp_clash is breaking because of this